README.rst

```diff
@@ -1,73 +1,44 @@
-more correct.
-
-So what?
-========
-
-You're probably thinking, "Who cares?" And you'd be right. I sincerely
-doubt using these customizations will make your life as a programmer
-even 1% more productive. 1% is nothing. You'd do better to buy a
-bigger monitor.
-
-So then, why be OCD about whitespace? It boils down to one essential
-thing: diffs like these.
-
-::
+The problem: spurious git diffs when whitespace changes happen.
 
-    $ git diff
-    diff --git a/foo.c b/foo.c
-    index 7be6eb3..8ba98ba 100644
-    --- a/foo.c
-    +++ b/foo.c
-    @@ -1,6 +1,6 @@
-     #include <stdio.h>
+When text and code does not have a canonical form, it's possible for
+someone to change from one form to another without intending to or
+noticing. As an example, someone can add or remove whitespace "after"
+the text on the end of a line without changing its meaning.
 
-    -int main(){
-    +int main(){  
-       printf("hello world\n");
-       return 0;
-     }
+A common solution to this is to automatically "clean" whitespace on
+every save. But this can introduce its own spurious diffs when working
+on a file that isn't already clean.
 
-What's going on in this diff? Take a few minutes to study it and figure out what's going on. What did I change? It doesn't look like I changed anything! But in fact there's a diff, so there must be some change, right?
+The solution: ``ethan-wspace``.
 
-The answer is that I either introduced or removed whitespace at the end of the line. ``git diff`` will highlight, in red, trailing whitespace like this that you added, but not that you took away. (P.S. ReStructuredText is removing the trailing whitespace from the example, but if you look at the rst source you will see that it is there.)
+How it works
+============
 
-Reviewing diffs with whitespace changes is annoying. Trying to guess what the whitespace changes were, so you can undo the whitespace changes, is downright tedious. Ideally you should never have to do these things.
+When ``ethan-wspace`` is activated in a buffer, it examines the
+whitespace in that buffer and does different things depending on
+whether that whitespace is clean or dirty.
 
-ethan-wspace can help.
+- If the whitespace is dirty, then ``ethan-wspace`` will highlight the
+  "errors", so you can be cognizant of where the whitespace is already
+  dirty. This can help you preserve the whitespace as-is, although it
+  does not prevent you from introducing new errors.
 
-What does it do?
-================
+- If the whitespace is already clean, then ``ethan-wspace`` will
+  insert hooks to clean this whitespace on save. This will ensure that
+  the whitespace remains clean, even if you introduce errors
+  yourself. Because the whitespace will be automatically cleaned,
+  there is no need to display whitespace specially, and no
+  highlighting is added.
 
-ethan-wspace takes the approach of *do no harm*. Specifically, when you open a file:
+It looks like this:
 
```

Comment:
What about adding something like the following to this sentence?
Where **ew:mnLT**  represents the current space-status of the buffer -- and a description of what each letter means-- (otherwise the reader will just see the red spaces and don't notice the importance of these letters)

Reply:
Do you refer to the section?

`There's also a mode line "lighter" for ethan-wspace which looks something like ew:mnLt. Lower case letters indicate categories of whitespace which will be cleaned before save; upper case letters indicate categories which are being highlit. You can click on a particular letter to change from one to the other.`

I think maybe it's better to have it a little bit upper and explain what each letter means. Reading this paragraph it's hard to the user to figure out what they mean.

Reply:
Sorry for the late response! yeah, I think a simple explanation at the beginning of the Readme of what the letters mean could be helpful for the user. Anyway when the buffer is dirty you are already highlighting the bad spaces which is the most important visual information/feedback you can give to the user. The minor mode letters are a minor issue IMHO.

Reply:
OK, I think this is enough of an improvement over what we have now that I'm going to merge it. If you have further suggestions for the text around the screenshot, please feel free to open a PR!